### PR TITLE
refactor(toBeIsoDateStringNear): remove momentjs dependency

### PR DESCRIPTION
**Pull request by mark-scott**
remove momentjs dependency from package.json
remove momentjs diff() usage
add millisecondsTo helper function
add isISOString() pattern matcher

### CODE REVIEW

**Comment by mark-scott** [comment]
Removed dependency to improve portability and maintainability, and negligible performance increase.

Refer to: https://github.com/you-dont-need/You-Dont-Need-Momentjs

**Comment by ncjones** [comment]
Hi @mark-scott thanks for the contribution.

Our current position WRT Momentjs is that it is still preferred for Nodejs workloads. For this library to drop Moment as a runtime dependency it would need to be backwards compatible with our existing tests which use it like:

```
expect(subject.expiresAt).toBeIsoDateStringNear(moment().add(1, 'hour'));
```

The `isISOString` function needs unit tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 1.0.3 - 2022-01-17
+### Removed
+- momentjs dependency
+### Added
```
Comment (ncjones) [inline CHANGELOG.md:11]: These are implementation details

**Comment by mark-scott** [comment]
> Hi @mark-scott thanks for the contribution.
> 
> Our current position WRT Momentjs is that it is still preferred for Nodejs workloads. For this library to drop Moment as a runtime dependency it would need to be backwards compatible with our existing tests which use it like:
> 
> ```
> expect(subject.expiresAt).toBeIsoDateStringNear(moment().add(1, 'hour'));
> ```
> 
> The `isISOString` function needs unit tests.

`moment()` can be passed into the `Date` constructor, or were you talking about matching functionality regarding addition/subtraction?

**Comment by ncjones** [comment]
@mark-scott I've pulled this into #75.

**Pull request closed without merging**